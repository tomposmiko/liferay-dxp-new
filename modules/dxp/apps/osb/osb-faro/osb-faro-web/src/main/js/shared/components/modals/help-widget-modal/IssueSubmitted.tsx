import ClayIcon from '@clayui/icon';
import Modal from 'shared/components/modal';
import React from 'react';
import {IHelpWidgetScreenProps} from './types';

const IssueSubmitted: React.FC<IHelpWidgetScreenProps> = ({onClose}) => (
	<>
		<Modal.Header
			onClose={onClose}
			title={Liferay.Language.get('issue-submitted')}
		/>

		<Modal.Body className='d-flex flex-column align-items-center'>
			<ClayIcon
				className='icon-root icon-size-xxxl my-5'
				symbol='ac-no-sites'
			/>

			<h3>{Liferay.Language.get('message-received')}</h3>

			<p className='description'>
				{Liferay.Language.get(
					'thanks-for-your-contribution-well-look-in-to-this-as-soon-as-possible'
				)}
			</p>
		</Modal.Body>
	</>
);

export default IssueSubmitted;
