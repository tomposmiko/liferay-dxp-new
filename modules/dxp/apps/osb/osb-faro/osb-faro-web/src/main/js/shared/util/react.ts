import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
