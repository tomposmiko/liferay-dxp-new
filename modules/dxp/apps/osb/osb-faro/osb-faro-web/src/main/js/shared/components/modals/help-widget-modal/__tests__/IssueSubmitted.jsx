import IssueSubmitted from '../IssueSubmitted';
import React from 'react';
import {noop} from 'lodash';
import {render} from '@testing-library/react';

jest.unmock('react-dom');

describe('IssueSubmitted', () => {
	it('should render', () => {
		const {container} = render(
			<IssueSubmitted onClose={noop} onNext={noop} />
		);

		expect(container).toMatchSnapshot();
	});
});
