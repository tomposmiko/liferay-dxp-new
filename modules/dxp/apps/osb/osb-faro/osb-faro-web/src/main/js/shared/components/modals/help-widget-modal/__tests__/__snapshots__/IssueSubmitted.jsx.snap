// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueSubmitted should render 1`] = `
<div>
  <div
    class="modal-header"
  >
    <h4
      class="modal-title"
    >
      Issue Submitted
    </h4>
    <button
      class="button-root btn btn-secondary close"
      type="button"
    >
      <svg
        class="icon-root lexicon-icon lexicon-icon-times"
        viewBox=""
      >
        <use
          xlink:href="/o/osb-faro-web/dist/sprite.svg#times"
        />
      </svg>
    </button>
  </div>
  <div
    class="modal-body d-flex flex-column align-items-center"
  >
    <svg
      class="icon-root lexicon-icon lexicon-icon-ac-no-sites my-5 icon-size-xxxl"
      viewBox=""
    >
      <use
        xlink:href="/o/osb-faro-web/dist/sprite.svg#ac-no-sites"
      />
    </svg>
    <h3>
      Message Received
    </h3>
    <p
      class="description"
    >
      Thanks for your contribution. We will look in to this as soon as possible.
    </p>
  </div>
</div>
`;
